java-client/src/integration-test/java/com/cumulocity/sdk/client/notification/RealtimeMultiSubNotificationIT.java

```diff
@@ -0,0 +1,194 @@
+import lombok.extern.slf4j.Slf4j;
+import org.awaitility.core.ConditionTimeoutException;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.*;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static com.cumulocity.sdk.client.notification.SubscriberBuilder.REALTIME;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+import static org.awaitility.Durations.ONE_SECOND;
+import static org.awaitility.Durations.TEN_SECONDS;
+
+@Slf4j
+public class RealtimeMultiSubNotificationIT extends JavaSdkITBase {
+    private static final int CONCURRENT_JOBS = 10;
+
+    InventoryApi inventoryApi = platform.getInventoryApi();
+    DeviceControlApi operationsApi = platform.getDeviceControlApi();
+
+    @Test
+    public void shouldNotReceiveDuplicatedOperations() throws InterruptedException, ExecutionException {
+        ExecutorService executor = Executors.newFixedThreadPool(CONCURRENT_JOBS);
+
+        List<Callable<Boolean>> jobs = new ArrayList<>();
+        for (int jobNo = 0; jobNo < CONCURRENT_JOBS; ++jobNo) {
+            jobs.add(new SubscriberJob(jobNo));
+        }
+
+        boolean failed = false;
+        for (Future<Boolean> result : executor.invokeAll(jobs)) {
+            if (!result.get()) {
+                failed = true;
+                break;
+            }
+        }
+
+        assertThat(failed)
+                .withFailMessage("Test failed. See the logs to investigate the issue.")
+                .isFalse();
+    }
+
+    private class SubscriberJob implements Callable<Boolean> {
+        private final int jobNo;
+        private final Subscriber<String, RealtimeMessage> subscriber;
+
+        public SubscriberJob(int jobNo) {
+            this.jobNo = jobNo;
+            this.subscriber = SubscriberBuilder.<String, RealtimeMessage>anSubscriber()
+                    .withParameters(platform)
+                    .withEndpoint(REALTIME)
+                    .withSubscriptionNameResolver(name -> name)
+                    .withDataType(RealtimeMessage.class)
+                    .build();
+        }
+
+        @Override
+        public Boolean call() throws InterruptedException {
+            ManagedObjectRepresentation device = inventoryApi.create(aMoDevice());
+
+            AtomicInteger opCounter = new AtomicInteger(0);
+            List<Subscription<String>> subscriptions = subscribe(device.getId(), opCounter);
+
+            operationsApi.create(aDeviceOperation(device.getId()));
+
+            try {
+                // check counter as fast as possible and then wait 1s to verify it didn't go up
+                await().atMost(TEN_SECONDS).until(() -> opCounter.get() == 1);
+                await().pollDelay(ONE_SECOND).atMost(TEN_SECONDS).until(() -> opCounter.get() == 1);
```

Comment:
Why two times? It won't be the case that the first awaits will check it every default time for 10 seconds and then the second await will do the same with just a one second poll instead of default.
If you like to assert quickly first maybe we should reduce this TEN_SECONDS?

Reply:
As I wrote in the comment the first time I'm waiting for `opCounter == 1` and the second time I'm waiting one more second to make sure it don't go up.